File: Gemini 1.5/Python/2-3.py

# 3. File Read and Write

def file_read_write():
  """
  This function reads content from one file and writes it to another file.
  """
  try:
    with open("input.txt", "r") as file_in:  # Open the input file in read mode
      content = file_in.read()  # Read the content of the file

    with open("output.txt", "w") as file_out:  # Open the output file in write mode
      file_out.write(content)  # Write the content to the output file

    print("File copied successfully!")
  except FileNotFoundError:
    print("Error: Input file not found!")


if __name__ == "__main__":
  file_read_write()

File: Gemini 1.5/C++/2-3.c
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main() {
    // Read from input.txt and write to output.txt
    ifstream inputFile("input.txt");
    ofstream outputFile("output.txt");

    if (!inputFile.is_open()) {
        cout << "Error opening input file!" << endl;
        return 1;
    }

    if (!outputFile.is_open()) {
        cout << "Error opening output file!" << endl;
        return 1;
    }

    string line;
    while (getline(inputFile, line)) {
        outputFile << line << endl;
    }

    inputFile.close();
    outputFile.close();

    cout << "File copied successfully!" << endl;

    return 0;
}

